gson/src/main/java/com/google/gson/Gson.java

```diff
@@ -925,7 +925,10 @@ private static void assertFullConsumption(Object obj, JsonReader reader) {
     boolean oldLenient = reader.isLenient();
     reader.setLenient(true);
     try {
-      reader.peek();
+      JsonToken token = reader.peek();
```

Comment:
Might be good to add a test for this.

Reply:
I agree with your other comment that removing `isEmpty` and `EOFException` block here and in `Stream::parse` should be another issue/PR.

Reply:
Ah sorry, you might have misunderstood me. My comments were not related to each other. Here in this method I think you can remove the `isEmpty` variable without introducing any behavior change. An `EOFException` will only be thrown either for empty documents or after `peek()` has succeeded at least once (and therefore the document is non-empty), see the `JsonReader` implementation. However with your change in lenient mode no EOFException will be thrown for empty documents. And because this method (`Gson.fromJson`) always makes the reader lenient, it should therefore be impossible that a EOFException is thrown for an empty document.

Reply:
I see what you are staying. So we can't remove the similar block in `Stream::parse` because the reader isn't necessarily lenient. I'll remove the block here.